android/samples/hello-triangle/app/build.gradle

```diff
@@ -1,9 +1,69 @@
-apply plugin: 'com.android.application'
+// This script accepts the following parameters:
+//
+// filament_tools_dir
+//     Path to the Filament distribution/install directory for desktop
+//     (produced by make/ninja install). This directory must contain bin/matc
+//
+// Example:
+//     ./gradlew -Pfilament_tools_dir=../../dist-release assembleDebug
 
+apply plugin: 'com.android.application'
 apply plugin: 'kotlin-android'
-
 apply plugin: 'kotlin-android-extensions'
 
+def filamentToolsPath = file("../../../../out/release/filament")
+if (project.hasProperty("filament_tools_dir")) {
+    filamentToolsPath = file("$filament_tools_dir")
+}
+
+// Custom task to compile material files using matc
+// This task handles incremental builds
+class MaterialCompiler extends DefaultTask {
+    @Input
+    File matcPath
+
+    @SuppressWarnings("GroovyUnusedDeclaration")
+    @InputDirectory
+    File inputDir
+
+    @OutputDirectory
+    File outputDir
+
+    @SuppressWarnings("GroovyUnusedDeclaration")
+    @TaskAction
+    void execute(IncrementalTaskInputs inputs) {
+        if (!inputs.incremental) {
+            project.delete(project.fileTree(outputDir).matching { include '*.filamat' })
+        }
+
+        inputs.outOfDate { InputFileDetails outOfDate ->
+            def file = outOfDate.file
+            project.exec {
+                executable "${matcPath}/bin/matc"
```

Comment:
This won't work on Windows, which is fine for now, but we might want to say that somewhere.

Reply:
Fixed!